File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/service/analytics/EnrollMerchantViewedAction.java
package com.worldventures.dreamtrips.social.ui.membership.service.analytics;

import com.worldventures.core.service.analytics.AdobeTracker;
import com.worldventures.core.service.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.BaseAnalyticsAction;

@AnalyticsEvent(action = "membership-enroll-merchant",
                trackers = AdobeTracker.TRACKER_KEY)
public class EnrollMerchantViewedAction extends BaseAnalyticsAction {

   @Attribute("member_id") final String memberId;

   @Attribute("view") final String view = "1";

   public EnrollMerchantViewedAction(String memberId) {
      this.memberId = memberId;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/service/analytics/EnrollMerchantViewedAction.kt
package com.worldventures.dreamtrips.social.ui.membership.service.analytics

import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.core.service.analytics.BaseAnalyticsAction
import com.worldventures.janet.analytics.AnalyticsEvent

@AnalyticsEvent(action = "membership-enroll-merchant", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class EnrollMerchantViewedAction(@field:Attribute("member_id") internal val memberId: String) : BaseAnalyticsAction() {

   @field:Attribute("view") internal val view = "1"
}
